modules/tinymce/src/core/main/ts/undo/Operations.ts

```diff
@@ -76,13 +76,13 @@ export const addUndoLevel = (editor: Editor, undoManager: UndoManager, index: In
 
   const args = { level, lastLevel, originalEvent: event };
 
-  editor.fire('AddUndo', args);
-
   if (index.get() > 0) {
     editor.setDirty(true);
     editor.fire('change', args);
   }
 
+  editor.fire('AddUndo', args);
```

Comment:
This is a potentially breaking change, so we can't do this sorry. We need to ensure the event chain is the same, so `change` needs to fire after `AddUndo`.

Reply:
I brought the `AddUndo` into the ```if/then``` so that the order is preserved.